core/crypto/client_tx.go

```diff
@@ -42,7 +46,11 @@ func (client *clientImpl) createDeployTx(chaincodeDeploymentSpec *obc.ChaincodeD
 	}
 
 	// Copy metadata from ChaincodeSpec
-	tx.Metadata = chaincodeDeploymentSpec.ChaincodeSpec.Metadata
+	tx.Metadata, err = getMetadata(chaincodeDeploymentSpec.GetChaincodeSpec(), tCert, attrs...)
```

Comment:
@andresgaragiola, May I suggest the following approach: Rather than changing the confidential transaction processing in the way is proposed in this PR, you could expose the internal method *tCert#GetKForAttribute* via the public *CertificateHandler* interface with a new method called something like *GetProofOfAttributeOwnership(attrName) ([]byte, error)*. This will give back the freedom to the application to decide how to embed that proof of ownership in the transaction metadata. This does not change the transaction structure and preserve the contract on the transaction metadata field. Thoughts?